### PR TITLE
Check that __clone, __construct, __destruct signatures omit return types

### PR DESCRIPTION
**Pull request by rightfold**
PHP complains when you have a return type declared for these three methods.

### CODE REVIEW

File: tests/MethodSignatureTest.php [hunk 1]
```diff
@@ -472,4 +472,26 @@ class B extends A {
...
+     * @expectedException        \Psalm\Exception\CodeException
+     * @expectedExceptionMessage MethodSignatureMustOmitReturnType
+     *
+     * @return                   void
+     */
+    public function testMustOmitReturnType()
+    {
+        $this->addFile(
+            'somefile.php',
+            '<?php
```
Comment (muglug) [inline tests/MethodSignatureTest.php]: this code can be added to the above data provider as

```php
'mustOmitReturnType' => [
    '<?php
        class A
        {
            public function __construct(): void
            {
            }
        }',
    'error_message' => 'MethodSignatureMustOmitReturnType'
]
```

File: src/Psalm/Checker/MethodChecker.php [hunk 2]
```diff
@@ -660,4 +661,35 @@ public static function compareMethods(
...
+        MethodStorage $method_storage,
+        CodeLocation $code_location
+    ) {
+        if ($method_storage->signature_return_type === null) {
+            return null;
+        }
+
+        $cased_method_name = $method_storage->cased_name;
+        switch ($cased_method_name) {
+        case '__clone':
```
Comment (muglug) [inline src/Psalm/Checker/MethodChecker.php]: Is this a common pattern? I'm used to indenting `case` statements

Comment (rightfold) [reply src/Psalm/Checker/MethodChecker.php]: Don't know. I can change it to a series of `===`/`||` or an `in_array` check instead.

Comment (muglug) [reply src/Psalm/Checker/MethodChecker.php]: no I just meant I'm used to seeing/writing

```php
switch ($foo) {
    case 'bar':
        // do something
}
```

Comment (rightfold) [reply src/Psalm/Checker/MethodChecker.php]: Non-indented case is pretty common, yes. But from a quick look at the code base I see that it indents them everywhere.

**Comment by muglug** [comment]
Thanks!

**Pull request merged**